### DISCUSSION TITLE
Silly question from a non-dev, will Terminal ever replace the classic Windows cmd?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/terminal#17814] [Q&A] [answered]
title!
be it in 25h2 or windows 12 or beyond, will it ever?

**Comment by lhecker** [comment, marked as answer]
It's already the default terminal application on Windows 11. When you run "cmd.exe" for instance it'll open Windows Terminal. There are still some rough edges, like how elevated (admin) console applications will still open in the classic console window, but perhaps we'll be able to address those in the future.

If you're asking if we'll ever remove the classic console from Windows, then the answer is that we currently don't plan to do so in the foreseeable future. The UI stack we're using for Windows Terminal (WinUI, D2D, D3D) is not as stable as the basic Win32 window and GDI text renderer as used in the classic console. As such it provides an excellent fallback in case something is not working right.

**Reply by ghost** [reply]
> It's already the default terminal application on Windows 11. When you run "cmd.exe" for instance it'll open Windows Terminal.

ooh I haven't heard anything about that! that basically answers what I was curious about


> remove the classic console from Windows

yeah I assumed it wouldn't get removed for at least a while due to compatibility, read my mind about it being used as fallback.

thanks for the answers!
